packages/block-editor/src/components/block-settings-menu-controls/edit-section-menu-item.js

```diff
@@ -71,6 +71,7 @@ export function EditSectionMenuItem( { clientId, onClose } ) {
 		} else {
 			// Use spotlight mode for unsynced patterns
 			editContentOnlySection( clientId );
+			selectBlock( clientId );
```

Comment:
Optional. I did this mainly to ensure the selected pattern appears in the list view. Also, when selecting a block in content only mode, it's already editable. "Edit section/pattern" thus selects the pattern. 

Open to suggestions. 😄